frontend/pulse3d/src/components/interactiveAnalysis/InteractiveAnalysisModal.js

```diff
@@ -956,121 +747,123 @@ export default function InteractiveWaveformModal({
+  const calculateYLimit = ({ y1, y2 }, windowedAnalysisBounds, markerX) => {
+    const { start: x1, end: x2 } = windowedAnalysisBounds;
+    const slope = (y2 - y1) / (x2 - x1);
+    return y1 + slope * (markerX - x1);
   };
 
-  const assignNewArr = (wellIndex, data, newValue, setState) => {
-    let newArr = [...data];
-    newArr[wellIndex] = newValue;
-    setState([...newArr]);
-  };
-
-  const setBothThresholdLines = (
-    newPeakY1 = peakValleyWindows[selectedWell].minPeaks,
-    newPeakY2 = peakValleyWindows[selectedWell].minPeaks,
-    newValleyY1 = peakValleyWindows[selectedWell].maxValleys,
-    newValleyY2 = peakValleyWindows[selectedWell].maxValleys,
-    well = selectedWell
-  ) => {
-    const wellIndex = twentyFourPlateDefinition.getWellIndexFromName(well);
-
-    assignNewArr(wellIndex, peakY1, newPeakY1, setPeakY1);
-    assignNewArr(wellIndex, peakY2, newPeakY2, setPeakY2);
-    assignNewArr(wellIndex, valleyY1, newValleyY1, setValleyY1);
-    assignNewArr(wellIndex, valleyY2, newValleyY2, setValleyY2);
-  };
-
-  const isNewY = (yToCompare, originalYArr) => {
-    return (
-      yToCompare &&
-      originalYArr.length !== 0 &&
-      originalYArr[wellIdx] &&
-      parseInt(yToCompare) !== parseInt(originalYArr[wellIdx])
-    );
-  };
+  const findInitialThresholdForFeature = (well, featureType) => {
+    const { coordinates, featuresForWells } = originalAnalysisData;
+    const { max, min } = timepointRange;
 
-  const filterFeature = (
-    featureType,
-    featureIndices,
-    startTime,
-    endTime,
-    wellCoords,
-    wellIndex = wellIdx,
-    featureThresholdY1,
-    featureThresholdY2
-  ) => {
-    const featureThresholdsY1 = {
-      peak: featureThresholdY1 || peakY1[wellIndex],
-      valley: featureThresholdY1 || valleyY1[wellIndex],
+    const compare = (a, b) => {
+      return featureType === "peaks" ? a < b : a > b;
     };
 
-    const featureThresholdsY2 = {
-      peak: featureThresholdY2 || peakY2[wellIndex],
-      valley: featureThresholdY2 || valleyY2[wellIndex],
-    };
+    const featureIdx = featureType === "peaks" ? 0 : 1;
 
-    return featureIndices.filter((idx) => {
-      // Can only filter if the data for this well has actually been loaded,
-      // which is not guaranteed to be the case with the staggered loading of data for each well
-      if (!wellCoords) return true;
-
-      const [featureMarkerX, featureMarkerY] = wellCoords[idx];
+    const wellSpecificFeatures = featuresForWells[well][featureIdx];
+    const wellSpecificCoords = coordinates[well];
 
-      const featureThresholdY = calculateYLimit(
-        featureThresholdsY1[featureType],
-        featureThresholdsY2[featureType],
-        featureMarkerX
-      );
+    // consider when no features were found in a well
+    if (wellSpecificCoords && wellSpecificFeatures.length > 0) {
+      // arbitrarily set to first feature
+      let idxOfThresholdValue = wellSpecificFeatures[0];
```

Comment:
You'll want to make sure that `wellSpecificFeatures[0]` is within the start and end window bounds.. I had changed this so that the features are filtered to within window bounds before doing any other comparisons. This is for the case that the force value at `wellSpecificFeatures[0]` is either greatest or least depending on which which feature you're comparing and is never replaced by an index within the window bounds